### PR TITLE
tray icon behaviour changes

### LINKED ISSUE #525: tray icon event handling
The tray icon behaves unintuitively (imho).

Doubleclicks should directly open the window.
The current context menu (on single left click) may better open on rightclick or be replaced by something useful (if doubleclick opens, we don't need a "show" entry) ... there is basically no limit on what to show like sync progress or the last arrived messages ...

### PR DESCRIPTION
**Pull request by nnice**
new behaviour:
- [x]  right click shows the contextmenu/popup
- [x] double left click opens and brings app to front
- [x] single left click brings to front (if opened) 

resolves #525

### CODE REVIEW

File: src/main/java/de/qabel/desktop/ui/tray/QabelTray.java [hunk 1]
```diff
@@ -108,22 +103,51 @@ public void mouseReleased(MouseEvent e) {
...
                 }
                 popup.setVisible(visible);
             }
 
+            private void bringAppToFront() {
+
+                Platform.runLater(() -> {
+                    if(!primaryStage.isShowing()){
+                        primaryStage.show();
+                    }else{
```
Comment (julianseeger) [inline src/main/java/de/qabel/desktop/ui/tray/QabelTray.java]: you can always call `toFront` (no need for the else)

**Comment by julianseeger** [comment]
functionality works for me except for the missing toFront on show

**Comment by audax** [comment]
|`src/main/java/de/qabel/desktop/ui/tray/QabelTrayImpl.java:18`|
|---|
AwtQabelTray could be a better name.

File: src/main/java/de/qabel/desktop/ui/tray/TrayIconMouseAdapter.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        this.qabelTray = qabelTray;
+        this.popup = qabelTray.popup;
+    }
+
+    @Override
+    public void mouseReleased(MouseEvent e) {
+        if (isDoubleLeftClick(e)) {
+            qabelTray.bringAppToFront();
+        } else if (isRightClick(e)) {
+            Point position = calculatePosition(e);
```
Comment (audax) [inline src/main/java/de/qabel/desktop/ui/tray/TrayIconMouseAdapter.java]: This doesn't check if it is a left click. A double right click would return true, too.

**Pull request merged**